scripts/git-hooks/prepare-commit-msg

```diff
@@ -0,0 +1,30 @@
+#!/bin/bash
+
+# Auto-wraps the commit message body to the length allowed by gitlint.
+# Does not special-case the commit title because it cannot be fixed automatically;
+# wrapping is not allowed there. If needed, gitlint's `commit-msg` hook will
+# give us a chance to edit interactively.
+#
+# This runs after the commit message has been entered (even if with "git commit -m ...")
+# but before gitlint runs in the commit-msg hook.
+
+COMMIT_MSG_FILE=$1
+_COMMIT_SOURCE=$2
+_SHA1=$3
+
+WIDTH=80
+
+# Re-wraps file $1 to $WIDTH characters per line, using prettier, in-place.
+rewrap() {
+  # Suppress the output of prettier, as it's overly verbose. Show only on errors.
+  prettier --no-config --no-editorconfig --prose-wrap always --print-width "$WIDTH" --write "$f" >/tmp/prettier.out \
+    || { cat /tmp/prettier.out; exit 1; }
+}
+
+# If gitlint is already happy, do nothing
+gitlint <"$COMMIT_MSG_FILE" >/tmp/gitlint.out 2>&1 && exit 0
+
+# If this is a fixup commit, do nothing
+<"$1" head -n1 | grep -q 'fixup!' && exit 0
```

Comment:
Does fixup mean anything special? Or is it similar in essence to the `nit` commits

Reply:
It's a feature of git. Fixup commits are commits with a special syntax that git recognizes and helps you use them to amend older commits. It's a very useful tool for keeping clean branches with meaningful commits (hint hint :)). This is a good intro, I have a very similar setup myself and use it many times a day; would recommend: https://jordanelver.co.uk/blog/2020/06/04/fixing-commits-with-git-commit-fixup-and-git-rebase-autosquash/
But there's a lot on the internet about fixup commits if the above is too terse.